sys/net/gnrc/application_layer/uhcpc/gnrc_uhcpc.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "net/fib.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/ipv6/nc.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/gnrc/netapi.h"
+#include "net/gnrc/netif.h"
+#include "net/ipv6/addr.h"
+#include "net/netdev2.h"
+#include "net/netopt.h"
+
+#include "net/uhcp.h"
+#include "log.h"
+#include "fmt.h"
+
+static kernel_pid_t gnrc_border_interface;
+static kernel_pid_t gnrc_wireless_interface;
+
+static void set_interface_roles(void)
+{
+    kernel_pid_t ifs[GNRC_NETIF_NUMOF];
+    size_t numof = gnrc_netif_get(ifs);
+    uint16_t dev_type;
+
+    for (size_t i = 0; i < numof && i < GNRC_NETIF_NUMOF; i++) {
+        kernel_pid_t dev = ifs[i];
+        int res = gnrc_netapi_get(dev, NETOPT_DEVICE_TYPE, 0, &dev_type, sizeof(dev_type));
+        if (res <= 0) {
+            dev_type = NETDEV2_TYPE_UNKNOWN;
+        }
+        if ((!gnrc_border_interface) && (dev_type == NETDEV2_TYPE_ETHERNET)) {
+            ipv6_addr_t addr, defroute;
+            gnrc_border_interface = dev;
+
+            ipv6_addr_from_str(&addr, "fe80::2");
+            gnrc_ipv6_netif_add_addr(dev, &addr, 64,
+                                     GNRC_IPV6_NETIF_ADDR_FLAGS_UNICAST);
+
+            ipv6_addr_from_str(&defroute, "::");
+            ipv6_addr_from_str(&addr, "fe80::1");
+            fib_add_entry(&gnrc_ipv6_fib_table, dev, defroute.u8, 16,
+                    0x00, addr.u8, 16, 0,
+                    (uint32_t)FIB_LIFETIME_NO_EXPIRE);
+        }
+        else if ((!gnrc_wireless_interface) && (dev_type != NETDEV2_TYPE_ETHERNET)) {
+            gnrc_wireless_interface = dev;
+        }
+
+        if (gnrc_border_interface && gnrc_wireless_interface) {
+            break;
+        }
+    }
+
+    LOG_INFO("uhcpc: Using %u as border interface and %u as wireless interface.\n", gnrc_border_interface, gnrc_wireless_interface);
```

Comment:
Can the LOG_INFO be set for `gnrc_uhcpc` instead? here and in the other logs. This helps to make the difference between this and `uhcpc` (I got confused while debugging)